### PR TITLE
Fix Feeds results, with integration tests

### PR DESCRIPTION
**Pull request by silverbucket**
This makes sure the feed fetching results, which is an array of activity objects, is passed back to the client. Includes integration tests to make sure any breakage is caught in the future.

### CODE REVIEW

File: test/integration.test.js
Comment (raucao) [inline test/integration.test.js]: it fetch? :)

The "it" blocks are for expectations ("it does xyz"). "describe" is for descriptions.

File: test/integration.test.js
Comment (raucao) [inline test/integration.test.js]: This is missing a describe block for what exactly is expected to send credentials.

File: packages/server/src/platform-instance.ts
Comment (raucao) [inline packages/server/src/platform-instance.ts]:
```suggestion
    if (this.config.persist && this.config.requireCredentials.includes(payload.type)) {
```

**Review by raucao — COMMENTED** [review]
Seems to work fine in general! :+1: 

Some nitpick comments about spec wording.

Aside from that, I think there's a bug with the object content. `brief_html` and `brief_text` contain the entire content from the source article, while `html` contains the abbreviated content. Shouldn't that be the other way around?

**Comment by silverbucket** [comment]
@raucao this is ready for review, you already did the heavy lifting in your first review, just need a last check and approval.

File: packages/platform-feeds/index.js
Comment (raucao) [inline packages/platform-feeds/index.js]: Why are these two different keys, when they both use the same data? The `brief_text` contains HTML in all the feeds I tried with the example. It's usually also the same as the full HTML (i.e. not brief), but I don't remember which feed I used before where I got different data in brief and full (but in reverse).

File: packages/platform-feeds/index.js
Comment (raucao) [inline packages/platform-feeds/index.js]: Same as above: why have a `text` key here that duplicates the (HTML) data from the other key?

**Review by raucao — COMMENTED** [review]
Left a couple more comments about the properties.

There's one more thing I just saw: there's a `date` and a `datenum` property in the AS `object` for when the feed item was published, but no `published` property for the activity itself. Shouldn't this data move to `published` to better conform with AS conventions? (Or at least be copied there?)

**Comment by raucao** [comment]
... btw, we could also merge as is and address the functional questions in new PRs later.

**Comment by silverbucket** [comment]
@raucao hey, for the `html` vs `text` fields, what do you think about stripping the html from the text fields? If the source text is not html, they'll both be plain text, but if the source is html, then there will be the text alternative.

As for the date, what do you suggest? that there be `published` property at the root of each activity stream (eg. one property for each AS in the array) `feed[2].published`. I guess that would be the most uniform as that's how we do it for XMPP yeah? I'm just wondering if we want to differentiate between the date of the post, vs the date the post was fetched, not sure.

**Comment by raucao** [comment]
> @raucao hey, for the `html` vs `text` fields, what do you think about stripping the html from the text fields? If the source text is not html, they'll both be plain text, but if the source is html, then there will be the text alternative.

I just had a quick look at the [RSS](https://www.rssboard.org/rss-specification) and [ATOM](https://validator.w3.org/feed/docs/atom.html) specs. In RSS, encoded HTML is *allowed* in `description`, but it's up to the client to parse. In ATOM, text elements like `summary` and `content` should have a `type` attribute that specifies if it's HTML or plaint text, or even something else.

I'm not sure if the Sockethub platform is the right place to implement content formats and conversions to be honest. But it should always send all possible information to the client, of course. It would certainly make client implementation more convenient if at least the content type information would be there for every item, but I'm not sure doubling or even quadrupling the content payload is the way to go.

So I would maybe propose to only send a `summary`/brief value, if it's actually different from the `description` or `content` (i.e. the feed is an ATOM feed and content is different from summary), and I would probably not send `_text` and `_html`, but just `content` plus `content_type` if it's available.

> As for the date, what do you suggest? that there be `published` property at the root of each activity stream (eg. one property for each AS in the array) `feed[2].published`. I guess that would be the most uniform as that's how we do it for XMPP yeah? I'm just wondering if we want to differentiate between the date of the post, vs the date the post was fetched, not sure.

The client knows when the post was fetched, but there are two possible dates from the feed that it may want: the date when the item was originally published (I believe that's what `published` is for in AS objects), and the date that the item was last updated. The latter is actually the only required date element for ATOM feed entries, while `published` is optional, and `pubDate` is optional in RSS, too.

**Comment by silverbucket** [comment]
@raucao OK I think I've addressed both of your comments regarding duplicate payload (i now also provide contentType) and the date property naming to published. Let me know what you think.

File: test/integration.test.js
Comment (raucao) [inline test/integration.test.js]: Looks like this shouldn't be commented...

Comment (silverbucket) [reply test/integration.test.js]: Yeah, in the process of trying to figure out what broke the integration test, but otherwise the PR is complete. (I won't merge until integration tests are working again)

**Comment by silverbucket** [comment]
Just FYI - the compliance is all passing, it's just that I consolidated the jobs `test`, `lint`, `coverage` into one `compliance` to reduce noise. the last time these ran was before `compliance` existed so it thinks they are missing.

**Review by raucao — CHANGES_REQUESTED** [review]
I'm sorry, but the [first feed I tried to fetch](https://kosmos.social/@raucao.rss) failed with the following error:

    "TypeError: Cannot read properties of null (reading 'trim')"

**Comment by raucao** [comment]
I'm wondering what `status: true` is:

Screenshot from 2022-10-12 12-05-30

**Comment by raucao** [comment]
One more thing: I don't think there should be a value for "updated" when the source feed does not contain that property.

**Comment by silverbucket** [comment]
> One more thing: I don't think there should be a value for `updated` when the source feed does not contain that property.

The `feedparser` API doesn't really distinguish, the `date` property is always either the same as `pubDate` or more recent (e.g. updated). So I've added a comparison, if they are the same it's set to undefined (and therefore not passed along).

**Review by raucao — APPROVED** [review]
Looking all good now! :clap: :rocket:

**Pull request merged**
